crates/protocol/src/types/primitive.rs

```diff
@@ -0,0 +1,89 @@
+//! A wrapper for the most intuitive implementations of the primitive types
+
+use bytes::{Buf, BufMut};
+
+use crate::{error, ProtocolType};
+
+#[derive(Debug, Clone)]
+pub struct Primitive<T>(pub T);
```

Comment:
This makes it more explicit as to how the type will be parsed: big endian, IEEE 754 floats, Rust-like sizes

Reply:
I think that since we already have our own trait, this wrapper only adds to the complication.

So I would prefer to not have this wrapper type on primitives. Whenever a wrapper is actually needed (e.g `VarInt`) we are free to use it.

Reply:
This also makes it impossible to give `LenVec` a special implementation just for byte arrays, so we'll just end up with another unnecessary type. Is this really worth it?

Reply:
Discussed on Discord